### PR TITLE
add pylint_django migrations check

### PR DESCRIPTION
**Pull request by gsnider2195**
### `nautobot_dev_example/migrations/0002_runpython.py`

```py
# Generated by Django 3.2.21 on 2024-02-16 08:32

from django.db import migrations


def runpython_noop(apps, schema_editor):
    pass


class Migration(migrations.Migration):
    dependencies = [
        ("nautobot_dev_example", "0001_initial"),
    ]

    operations = [migrations.RunPython(code=runpython_noop)]
```

### Output

```sh
$ inv pylint
Running docker compose command "ps --services --filter status=running"
>>>> Executing external compose provider "/usr/libexec/docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

Running docker compose command "run --rm --entrypoint='pylint --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml nautobot_dev_example' nautobot"
>>>> Executing external compose provider "/usr/libexec/docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

[+] Creating 2/0
 ✔ Container nautobot-dev-example-db-1     Running                                                                                                      0.0s 
 ✔ Container nautobot-dev-example-redis-1  Running                                                                                                      0.0s 
18:11:05.133 INFO    nautobot             __init__.py                              setup() : Nautobot initialized!

------------------------------------
Your code has been rated at 10.00/10

Running docker compose command "ps --services --filter status=running"
>>>> Executing external compose provider "/usr/libexec/docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

Running docker compose command "run --rm --entrypoint='pylint --verbose --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml --load-plugins=pylint_django.checkers.migrations --disable=all --enable=new-db-field-with-default,missing-backwards-migration-callable nautobot_dev_example.migrations' nautobot"
>>>> Executing external compose provider "/usr/libexec/docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

[+] Creating 2/0
 ✔ Container nautobot-dev-example-redis-1  Running                                                                                                      0.0s 
 ✔ Container nautobot-dev-example-db-1     Running                                                                                                      0.0s 
18:11:17.139 INFO    nautobot             __init__.py                              setup() : Nautobot initialized!
Using config file pyproject.toml
************* Module nautobot_dev_example.migrations.0002_runpython
nautobot_dev_example/migrations/0002_runpython.py:15:18: W5197: Always include backwards migration callable (missing-backwards-migration-callable)

-----------------------------------
Your code has been rated at 9.33/10

Error: executing /usr/libexec/docker/cli-plugins/docker-compose --project-name nautobot-dev-example --project-directory /home/gary/github/nautobot/nautobot-app-dev-example/development -f /home/gary/github/nautobot/nautobot-app-dev-example/development/docker-compose.base.yml -f /home/gary/github/nautobot/nautobot-app-dev-example/development/docker-compose.redis.yml -f /home/gary/github/nautobot/nautobot-app-dev-example/development/docker-compose.postgres.yml -f /home/gary/github/nautobot/nautobot-app-dev-example/development/docker-compose.dev.yml run --rm --entrypoint=pylint --verbose --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml --load-plugins=pylint_django.checkers.migrations --disable=all --enable=new-db-field-with-default,missing-backwards-migration-callable nautobot_dev_example.migrations nautobot: exit status 4
$ echo $?
1
$ 
```

### `nautobot_dev_example/migrations/0002_runpython.py`

```py
# Generated by Django 3.2.21 on 2024-02-16 08:32

from django.db import migrations


def runpython_noop(apps, schema_editor):
    pass


class Migration(migrations.Migration):
    dependencies = [
        ("nautobot_dev_example", "0001_initial"),
    ]

    operations = [migrations.RunPython(code=runpython_noop, reverse_code=migrations.RunPython.noop)]
```

### Output

```sh
$ inv pylint
Running docker compose command "ps --services --filter status=running"
>>>> Executing external compose provider "/usr/libexec/docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

Running docker compose command "run --rm --entrypoint='pylint --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml nautobot_dev_example' nautobot"
>>>> Executing external compose provider "/usr/libexec/docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

[+] Creating 2/0
 ✔ Container nautobot-dev-example-db-1     Running                                                                                                      0.0s 
 ✔ Container nautobot-dev-example-redis-1  Running                                                                                                      0.0s 
18:14:40.335 INFO    nautobot             __init__.py                              setup() : Nautobot initialized!

------------------------------------
Your code has been rated at 10.00/10

Running docker compose command "ps --services --filter status=running"
>>>> Executing external compose provider "/usr/libexec/docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

Running docker compose command "run --rm --entrypoint='pylint --verbose --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml --load-plugins=pylint_django.checkers.migrations --disable=all --enable=new-db-field-with-default,missing-backwards-migration-callable nautobot_dev_example.migrations' nautobot"
>>>> Executing external compose provider "/usr/libexec/docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

[+] Creating 2/0
 ✔ Container nautobot-dev-example-db-1     Running                                                                                                      0.0s 
 ✔ Container nautobot-dev-example-redis-1  Running                                                                                                      0.0s 
18:14:52.505 INFO    nautobot             __init__.py                              setup() : Nautobot initialized!
Using config file pyproject.toml

------------------------------------
Your code has been rated at 10.00/10

$ echo $?
0
$ 
```

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -709,8 +709,21 @@ def hadolint(context):
...
+    exit_code = 0
+
     command = 'pylint --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml nautobot_dev_example'
-    run_command(context, command)
+    if not run_command(context, command, warn=True):
+        exit_code = 1
+
+    # run the pylint_django migrations checkers on the migrations directory
+    migrations_pylint = (
+        'pylint --verbose --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml '
```
Comment (smk4664) [inline tasks.py:720]: Do we need the `--verbose`? I see you added that to the migrations_pylint command, but not to the other pylint command.

**Review by smk4664 — DISMISSED** [review]
This looks great, just a question on the verbose.

File: tasks.py (hunk 1 above)
Comment (gsnider2195) [reply tasks.py:720]: Oh good catch! Core uses `--verbose` so I think we should add it to both commands.

**Review by bryanculver — APPROVED** [review]

**Review by smk4664 — APPROVED** [review]
Even better than before! Thanks for fixing the Verbose part.

**Pull request merged**